samples/bluetooth/direct_test_mode/src/fem/nrf21540.c

```diff
@@ -434,14 +460,38 @@ int nrf21540_init(void)
 
 	k_busy_wait(NRF21540_PD_PG_TIME_US);
 
+	atomic_set(&state, NRF21540_STATE_READY);
+
+	return 0;
+}
+
+int nrf21540_powerdown(void)
+{
+	int err;
+
+	if (!atomic_cas(&state, NRF21540_STATE_READY, NRF21540_STATE_BUSY)) {
+		return -EBUSY;
+	}
+
+	/* Power-down nRF21540 */
+	err = gpio_pin_set(nrf21540_cfg.pdn.port, nrf21540_cfg.pdn.pin, 0);
+	if (err) {
+		return err;
```

Comment:
`state` will remain `NRF21540_STATE_BUSY` if an error occurs here. Won't that be a problem? Similar story in `nrf21540_power_up()`.

Reply:
It could be. I modified the logic of those function. Now they check if error occurs and sets a valid state before return.